src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -823,26 +826,69 @@ private async Task CreateServicesAsync(CancellationToken cancellationToken = def
                 return;
             }
 
-            // We do not specify the initial list version, so the watcher will give us all updates to Service objects.
-            IAsyncEnumerable<(WatchEventType, Service)> serviceChangeEnumerator = kubernetesService.WatchAsync<Service>(cancellationToken: cancellationToken);
-            await foreach (var (evt, updated) in serviceChangeEnumerator)
+            var withTimeout = new TimeoutStrategyOptions()
             {
-                if (evt == WatchEventType.Bookmark) { continue; } // Bookmarks do not contain any data.
+                Timeout = _options.Value.ServiceStartupWatchTimeout
+            };
+
+            var tryTwice = new RetryStrategyOptions()
+            {
+                BackoffType = DelayBackoffType.Constant,
+                MaxDelay = TimeSpan.FromSeconds(1),
```

Comment:
Is this enough time. Based on the settings here it will trice once, then wait a second and try again, then again. So if it isn't responsive after what 3 seconds its going to give up?

This means that the timeout value specified in the options won't be that effective since these hard coded settings means that a value have 2-3 seconds won't be effective.